### PR TITLE
AllenNLP update/support for whole-word-masking BERT [WIP]

### PR DESCRIPTION
**Pull request by sleepinyourhat**
Addresses half of #730.

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ dependencies:
     - pandas=0.23.0
     - pytz=2017.3
     - pip:
-        - allennlp==0.8.1
+        - allennlp==0.8.4
```
Comment (sleepinyourhat) [inline environment.yml]: For compatibility with the latest pytorth_pretrained_bert. Doesn't break anything else.

**Comment by sleepinyourhat** [comment]
Also resolves #709.

**Comment by sleepinyourhat** [comment]
Relevant: https://github.com/huggingface/pytorch-pretrained-BERT/issues/659#issuecomment-506210983

File: environment.yml (hunk 1 above)
Comment (sleepinyourhat) [reply environment.yml]: False alarm. This does break stuff.

**Comment by sleepinyourhat** [comment]
@W4ngatang, I may need your help on this one. It looks like upgrading AllenNLP 0.8.1 => 0.8.4 breaks our patched version of MultiLabelField, and I don't think I understand the use case for the patch well enough to fix it. 

Mind taking a look? We can wait until after 1.0 if necessary, but it'd be nice to be up to date from launch.

**Comment by W4ngatang** [comment]
Can take a look

File: environment.yml (hunk 1 above)
Comment (pruksmhc) [reply environment.yml]: Has something been done to make sure this does not break things?

**Comment by pruksmhc** [comment]
What's the status of this @sleepinyourhat ?

**Comment by sleepinyourhat** [comment]
See last couple of comments—it's in @W4ngatang's hands.

**Comment by sleepinyourhat** [comment]
Sooner is better, but doesn't have to be in 1.0.

**Comment by W4ngatang** [comment]
@sleepinyourhat do you have a MWE for the break with multilabel field? I'm not seeing it running demo.

**Comment by W4ngatang** [comment]
Oh, looks like WSC uses it. NVM.

**Comment by W4ngatang** [comment]
I'm still not getting the error re: `MultiLabelField` using WSC (`winograd-coreference`) to test

**Comment by sleepinyourhat** [comment]
Hrm—it was in a slightly odd setup. If you can't reproduce it and WWM BERT
works, feel free to merge...

**Comment by W4ngatang** [comment]
Caveat: PPB on pip doesn't have the whole word masking models, so users that want to use them will have to install PPB through the git repo for a while.

**Comment by W4ngatang** [comment]
Oh you said that haha, I just can't read.
